### DISCUSSION TITLE
Email 2FA is disabled

### DISCUSSION THREAD

**Discussion by acagastya** [dani-garcia/vaultwarden#1788] [Q&A] [answered]
Hi, I am trying to set up email 2FA for the self-hosted bitwarden, but I get "Email 2FA is disabled" when I click "Send email" to "1. Enter the email that you wish to receive verification codes" in `/#/settings/two-factor`.  Any idea how I can fix that?

**Comment by BlackDex** [comment, marked as answer]
Looks like email isn't configured.
See: https://github.com/dani-garcia/vaultwarden/wiki/SMTP-configuration

**Reply by acagastya** [reply]
I am not running this using docker, so should I place this in the `.env` file instead, @BlackDex?
```
SMTP_HOST=localhost
SMTP_FROM=noreply@bitwarden.example.com
SMTP_SSL=false
SMTP_PORT=25
```

I tried that.  The bitwarden account is `foo@bar.com`.  When I enter that to receive code, I get `SMTP 5xx error: ["5.1.1 <foo@bar.com>... User unknown"]`.  But when I enter `foo@outlook.com`, that seems to work fine.  I don't know why that would be.

**Reply by BlackDex** [reply]
Within the .env file should be more then enough (unless you also used the /admin interface to store changes, which then would lead to the stored config override those settings).

I don't know what you use as your e-mail provider and if they support outgoing e-mail.
Some people use gmail, some have there own SMTP server somewhere.

**Reply by acagastya** [reply]
I think by default it uses `sendmail`?  I am facing no problem sending non-TLS emails to `foo@outlook.com`, but `foo@example.com` gives an error.  Unsure why it would not do that.

**Reply by BlackDex** [reply]
I would check the logs of you sendmail too see why.
You can also (temporarily) enable SMTP_DEBUG=true to have some verbose output during the sending.
You can use the /admin portal to send test emails (if you were not already doing that).
